### ISSUE TITLE
Build failure: python312Packages.keep

### ISSUE THREAD

**Issue by mknudsen** [NixOS/nixpkgs#353771]
### Steps To Reproduce

1. run `nix-shell -p python312Packages.keep`

### Build log

```shell
this derivation will be built:
  /nix/store/g1hflqj5pqpk6y79jxg4l2dxh8rbmgkl-python3.12-keep-2.11.drv
building '/nix/store/g1hflqj5pqpk6y79jxg4l2dxh8rbmgkl-python3.12-keep-2.11.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Running phase: unpackPhase
unpacking source archive /nix/store/9zvlfnp4v0386s5j26zyhbw2k0az9622-keep-2.11.tar.gz
source root is keep-2.11
setting SOURCE_DATE_EPOCH to timestamp 1722455844 of file keep-2.11/pyproject.toml
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "/private/tmp/nix-build-python3.12-keep-2.11.drv-0/keep-2.11/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

File "/nix/store/2djmffykchgm4q4j7ylv7xgkg441mp2j-python3-3.12.7/lib/python3.12/tokenize.py", line 449, in open
    buffer = _builtin_open(filename, 'rb')
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
error: builder for '/nix/store/g1hflqj5pqpk6y79jxg4l2dxh8rbmgkl-python3.12-keep-2.11.drv' failed with exit code 1;
       last 25 log lines:
       > Using setuptoolsBuildPhase
       > Sourcing pypa-install-hook
       > Using pypaInstallPhase
       > Sourcing python-imports-check-hook.sh
       > Using pythonImportsCheckPhase
       > Sourcing python-namespaces-hook
       > Sourcing python-catch-conflicts-hook.sh
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/9zvlfnp4v0386s5j26zyhbw2k0az9622-keep-2.11.tar.gz
       > source root is keep-2.11
       > setting SOURCE_DATE_EPOCH to timestamp 1722455844 of file keep-2.11/pyproject.toml
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > Executing setuptoolsBuildPhase
       > Traceback (most recent call last):
       >   File "/private/tmp/nix-build-python3.12-keep-2.11.drv-0/keep-2.11/nix_run_setup", line 8, in <module>
       >     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
       >                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/2djmffykchgm4q4j7ylv7xgkg441mp2j-python3-3.12.7/lib/python3.12/tokenize.py", line 449, in open
       >     buffer = _builtin_open(filename, 'rb')
       >              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       > FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
       For full logs, run 'nix-store -l /nix/store/g1hflqj5pqpk6y79jxg4l2dxh8rbmgkl-python3.12-keep-2.11.drv'.
```

### Notify maintainers

@risicle

### Metadata

```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Kek5chen** [comment]
The issue here seems to be with `python312Packages.keep`, not `howdoi`.

**Comment by dotlambda** [comment]
https://github.com/NixOS/nixpkgs/commit/361a131eca9a0b0f3f985e5d3fb8c5e5d6df488c was part of https://github.com/NixOS/nixpkgs/pull/345326 which is automated and doesn't test all builds

**Comment by mweinelt** [comment]
```diff
diff --git a/pkgs/development/python-modules/keep/default.nix b/pkgs/development/python-modules/keep/default.nix
index 6023df467420..da6210bcb337 100644
--- a/pkgs/development/python-modules/keep/default.nix
+++ b/pkgs/development/python-modules/keep/default.nix
@@ -2,8 +2,9 @@
   lib,
   buildPythonPackage,
   fetchPypi,
+  flit-core,
   pygithub,
-  terminaltables,
+  terminaltables3,
   click,
   requests,
 }:
@@ -11,17 +12,19 @@
 buildPythonPackage rec {
   pname = "keep";
   version = "2.11";
-  format = "setuptools";
+  pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-Brwvu/Zevr8sOE3KAwakDDzVMc2VoFxIb1orXAes2U0=";
+    hash = "sha256-Brwvu/Zevr8sOE3KAwakDDzVMc2VoFxIb1orXAes2U0=";
   };
 
-  propagatedBuildInputs = [
+  build-system = [ flit-core ];
+
+  dependencies = [
     click
     requests
-    terminaltables
+    terminaltables3
     pygithub
   ];
 ```

and then someone needs to package terminaltables3.

**Comment by Kek5chen** [comment]
will take a look

**Comment by Kek5chen** [comment]
Fixed in #354697
